File: 1/main.lua
local file = io.open("1/input.txt", "r")

if file == nil then
	return
end

local rotation = 50
local result = 0

for line in file:lines() do
	local distance = line:sub(2, -1)

	if line:sub(1, 1) == "L" then
		rotation = (rotation - tonumber(distance)) % 100
	else
		rotation = (rotation + tonumber(distance)) % 100
	end

	if rotation == 0 then
		result = result + 1
	end
end

print(result)

File: 1/main.ml
let modulo x y =
  let r = x mod y in
  if r >= 0 then r else r + y

let f (rotation, result) data =
  let distance = String.sub data 1 (String.length data - 1) |> int_of_string in
  let m = if String.get data 0 = 'L' then -1 else 1 in
  let r = modulo (rotation + (distance * m)) 100 in
  (r, if r = 0 then result + 1 else result)

let _, res =
  In_channel.with_open_text "1/input.txt" In_channel.input_lines
  |> List.fold_left f (50, 0)
;;

print_int res
